internal/cmd/ssh.go

```diff
@@ -198,6 +217,19 @@ func setupDestinationSSHConfig(ctx context.Context, cli *CLI, destination *api.D
 	return nil
 }
 
+// mkdirAll creates a directory and all its parents with mode 0o700, or returns
+// an error. mkdirAll is a wrapper around os.MkdirAll to provide a better error
+// message when the path already exists as a file.
+func mkdirAll(path string) error {
+	err := os.MkdirAll(path, 0o700)
+	pathError := &fs.PathError{}
+	if errors.As(err, &pathError) && errors.Is(pathError.Err, syscall.ENOTDIR) {
+		return fmt.Errorf("failed to create directory %v, the path already exists as a regular file",
+			pathError.Path)
+	}
+	return err
```

Comment:
I think this error should still be captured somewhere even if its an error other than one for a dir that already exists.

Reply:
For sure, I think both these errors should get printed.

The only time we hide errors are in the cases above that wrap `errBeforeDestinationMatch`, and those are printed as debug logs, so we can get them to show up when we need to.